mmv1/third_party/terraform/.teamcity/components/inputs/diff_check.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+/*
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// This file is controlled by MMv1, any changes made here will be overwritten
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+)
+
+func main() {
+	repo := os.Args
+	services := fmt.Sprintf("../../../%v/services/...", repo[1])
+	fmt.Println(services)
+	cmd := exec.Command("go", "list", services)
+	stdout, err := cmd.Output()
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+
+	pattern := regexp.MustCompile(`github\.com\/hashicorp\/terraform-provider-(google|google-beta)\/(google|google-beta)\/services\/(?P<service>\w+)`)
```

Comment:
If "google" / "google-beta" is obtained as an argument, or can be inferred from other arguments, that could be used to produce a slightly simpler regex.